eth/downloader/fetchers_concurrent.go

```diff
@@ -265,9 +280,9 @@ func (d *Downloader) concurrentFetch(queue typedQueue) error {
 			// cancel it, so it's not considered in-flight anymore, but keep
 			// the peer marked busy to prevent assigning a second request and
 			// overloading it further.
-			delete(requests, req.Peer)
+			delete(pending, req.Peer)
+			stales[req.Peer] = req
```

Comment:
With this change here, afaict the comment is now misleading. We don't cancel it so it's not considered in-flight anymore, since we no longer call `req.Close()` on it. I suppose the caller keeps waiting for the reply, even though this internally has been moved to stale? And after a little while longer, the peer will be rejected, prompting the stale request to be `Close`d and rescheduled?